query/src/datasources/table/github/github_table.rs

```diff
@@ -0,0 +1,159 @@
+    }
+
+    async fn get_repos_array(&self, limit: u8) -> Result<Vec<Series>> {
+        let repos = octocrab::instance()
+            .orgs(&self.table_info.name)
+            .list_repos()
+            .repo_type(params::repos::Type::Sources)
+            .sort(params::repos::Sort::Pushed)
+            .direction(params::Direction::Descending)
+            .per_page(limit)
+            .send()
+            .await?;
+
+        let mut repo_name_array: Vec<Vec<u8>> = Vec::new();
+        let mut language_array: Vec<Vec<u8>> = Vec::new();
+        let mut license_array: Vec<Vec<u8>> = Vec::new();
+        let mut star_count_array: Vec<u32> = Vec::new();
+        let mut forks_count_array: Vec<u32> = Vec::new();
+        let mut watchers_count_array: Vec<u32> = Vec::new();
+        let mut open_issues_count_array: Vec<u32> = Vec::new();
+        let mut subscribers_count_array: Vec<u32> = Vec::new();
+
+        let mut iter = repos.items.iter();
+        for repo in &mut iter {
+            repo_name_array.push(repo.name.clone().into());
+            language_array.push(
+                repo.language
+                    .as_ref()
+                    .and_then(|l| l.as_str())
+                    .unwrap_or("No Language")
+                    .as_bytes()
+                    .to_vec(),
+            );
+            license_array.push(
+                repo.license
+                    .as_ref()
+                    .map(|l| l.key.as_str())
+                    .unwrap_or("No license")
+                    .as_bytes()
+                    .to_vec(),
+            );
+            star_count_array.push(repo.stargazers_count.unwrap_or(0));
+            forks_count_array.push(repo.forks_count.unwrap_or(0));
+            watchers_count_array.push(repo.watchers_count.unwrap_or(0));
+            open_issues_count_array.push(repo.open_issues_count.unwrap_or(0));
+            subscribers_count_array.push(repo.subscribers_count.unwrap_or(0) as u32);
+        }
+
+        Ok(vec![
+            Series::new(repo_name_array),
+            Series::new(language_array),
+            Series::new(license_array),
+            Series::new(star_count_array),
+            Series::new(forks_count_array),
+            Series::new(watchers_count_array),
+            Series::new(open_issues_count_array),
+            Series::new(subscribers_count_array),
+        ])
+    }
+}
+
+#[async_trait::async_trait]
+impl Table for GithubTable {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn get_table_info(&self) -> &TableInfo {
+        &self.table_info
+    }
+
+    async fn read(
+        &self,
+        _ctx: DatabendQueryContextRef,
+        plan: &ReadDataSourcePlan,
+    ) -> Result<SendableDataBlockStream> {
+        let limit = plan
+            .push_downs
+            .as_ref()
+            .and_then(|extras| extras.limit)
+            .unwrap_or(100) as u8;
+
+        let arrays = self.get_repos_array(limit).await?;
```

Comment:
This limit did not respect sort plan.

Reply:
thank you for reminding of that, I will fix it later.